src/review09_10/Puzzle07.java

```diff
@@ -0,0 +1,45 @@
+package review09_10;
+
+import java.util.Comparator;
+import java.util.TreeSet;
+
+public class Puzzle07 {
+
+    // What does this program print?
+    //
+    // A. [10, 30, 300, 400, 5000]
+    // B. [300, 10, 30, 5000, 400]
+    // C. [400, 300, 10, 30, 5000]
+    // D. [400, 5000, 30, 10, 300]
+    // E. [5000, 30, 10, 300, 400]
+    // F. [5000, 400, 300, 30, 10]
+    // G. The order is undefined
+
+    record Animal(String name, int weight) implements Comparable<Animal>, Comparator<Animal> {
+        @Override
+        public int compareTo(Animal other) {
+            return other.name.compareTo(this.name);
+        }
+
+        @Override
+        public int compare(Animal a1, Animal a2) {
+            return Integer.compare(a1.weight, a2.weight);
+        }
+
+        @Override
+        public final String toString() {
+            return "" + weight;
+        }
+    }
+
+    public static void main(String[] args) {
+        TreeSet<Animal> a = new TreeSet<>();
```

Comment:
Deze wordt natuurlijk nóg leuker als we aan de constructor een `new Animal(...)` meegeven als Comparator!